Tracking3D/Tracking3DLib/TreeNode.py

```diff
@@ -0,0 +1,141 @@
+        import logging
+
+        try:
+            logging.info(f"Searching for {self.name} transforms")
+            newSequenceNode = slicer.util.getNode(f"{self.name}_transform_sequence")
+        except slicer.util.MRMLNodeNotFoundException:
+            try:  # Loading the sequence transforms from seq and tfm files can be wacky
+                logging.info(f"Searching for {self.name} transforms")
+                newSequenceNode = slicer.util.getNode(
+                    f"{self.name}_transform_sequence-{self.name}_transform_sequence-Seq"
+                )
+            except slicer.util.MRMLNodeNotFoundException:
+                logging.info(f"Transforms not found, Initializing {self.name}")
+                newSequenceNode = self.autoscoperLogic.createSequenceNodeInBrowser(
+                    f"{self.name}_transform_sequence", self.ctSequence
+                )
+                nodes = []
+                for i in range(self.ctSequence.GetNumberOfDataNodes()):
+                    curTfm = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLLinearTransformNode", f"{self.name}-{i}")
+                    newSequenceNode.SetDataNodeAtValue(curTfm, f"{i}")
+                    nodes.append(curTfm)
+                [slicer.mrmlScene.RemoveNode(node) for node in nodes]
+
+                # Bit of a strange issue but the browser doesn't seem to update unless it moves to a new index,
+                # so we force it to update here
+                self.autoscoperLogic.getItemInSequence(newSequenceNode, 1)
+                self.autoscoperLogic.getItemInSequence(newSequenceNode, 0)
+
+                slicer.app.processEvents()
+        return newSequenceNode
+
+    def _applyTransform(self, transform: slicer.vtkMRMLTransformNode, idx: int) -> None:
+        """Applies and hardends a transform node to the transform in the sequence at the provided index."""
+        if idx < self.transformSequence.GetNumberOfDataNodes():
+            current_transform = self.autoscoperLogic.getItemInSequence(self.transformSequence, idx)[0]
+            current_transform.SetAndObserveTransformNodeID(transform.GetID())
+            current_transform.HardenTransform()
+
+    def getTransform(self, idx: int) -> slicer.vtkMRMLTransformNode:
+        """Returns the transform at the provided index."""
+        if idx < self.transformSequence.GetNumberOfDataNodes():
+            return self.autoscoperLogic.getItemInSequence(self.transformSequence, idx)[0]
+        return None
+
+    def setTransformFromNode(self, transform: slicer.vtkMRMLLinearTransformNode, idx: int) -> None:
+        """Sets the transform for the provided index."""
+        if idx < self.transformSequence.GetNumberOfDataNodes():
+            mat = vtk.vtkMatrix4x4()
+            transform.GetMatrixTransformToParent(mat)
+            current_transform = self.autoscoperLogic.getItemInSequence(self.transformSequence, idx)[0]
+            current_transform.SetMatrixTransformToParent(mat)
```

Comment:
Should a warning be logged if `idx >= self.transformSequence.GetNumberOfDataNodes()` ?

Reply:
Same comment applies to functions below